src/tensor/traits.ts

```diff
@@ -61,6 +62,214 @@ export const nullLikeTraitValues = [
 ];
 export const NONE_TRAIT_VALUE = 'None';
 
+// Synth traits are not stored on-chain
+enum SynthTrait {
+  DIGITS_ONLY = "Digits Only",
+  LETTERS_ONLY = "Letters Only",
+  PALINDROME = "Palindrome",
+  EMOJI = "Emoji",
+  LANGUAGE = "Language",
+  CATEGORY = "Category",
+  LETTER_COUNT = "Letter Count",
+}
+
+enum TraitType {
+  DOMAIN = "domain",
+}
+
+interface SynthTraitArg {
+  name?: string;
+  currentTraits?: Set<string>;
+}
+
+export const DOMAIN_MCC_LIST = [
+  "E5ZnBpH9DYcxRkumKdS4ayJ3Ftb6o3E8wSbXw4N92GWg",
+  "86deDknZeDhko46gB8SqK7rYc5HnSBjKDvo6Mi7viYS9",
+  "6bsj8ybPa9xsc6pcAme4x6LvhKvtCmgA4TwwG4qtFw5Z",
+  "GYLiNNu4pqL6QvZKYHW2EMoibVFm2aVJsPHpUVLcU6pL",
+  "7yQYe84W7a5VgNvtRzsvy7mPRed5gmL9HnvJfsbPWK9J",
+];
+
+export const languageRegex = {
+  English: /[a-zA-Z]/,
+  Arabic: /[\u0600-\u06FF]/,
+  Chinese: /[\u4e00-\u9FFF]/,
+  Cyrillic: /[\u0400-\u04FF]/,
+  Hindi: /[\u0900-\u097F]/,
+  Japanese: /[\u3040-\u309F\u30A0-\u30FF]/,
+  Korean: /[\uAC00-\uD7AF]/,
+};
+
+// Use this when emojis are in strings because emojis aren't always the same size
+const visibleLength = (str: string) => [...new Intl.Segmenter().segment(str)].length;
+
+const isNumeric = (str: string) => /^\d+$/.test(str);
+const isAlpha = (str: string) => /^[a-zA-Z]+$/.test(str);
+const emojiLength = (str: string) =>
+  [...new Intl.Segmenter().segment(str)]
+    .map((emoji) => emoji.segment)
+    .filter((char) => /\p{Extended_Pictographic}/gu.test(char)).length;
+
+const isPalindrome = (str: string) =>
+  str ===
+  [...new Intl.Segmenter().segment(str)]
+    .map((emoji) => emoji.segment)
+    .reverse()
+    .join("");
+
+const determineNumberClub = (name: string) => {
+  const number = parseInt(name);
+  if (number <= 1000) {
+    return "1k Club";
+  } else if (number <= 10000) {
+    return "10k Club";
+  } else if (number <= 100000) {
+    return "100k Club";
+  }
+};
+
+const syntheticTraits = [
```

Comment:
When we need to add in synthetic traits that may not be related to domains in the future, this lets us add it easily